design-documents/storefront/configuraiton-propagation.md

```diff
@@ -0,0 +1,65 @@
+# Configuration Propagation from Admin Panel to Storefront
+
+## Problem Statement
+
+Existing Magento monolith is responsible for the whole cycle of data and user workflow.
+Magento Admin Panel contains configuration settings for all areas of the application, including Admin Panel behavior, storefront UI, etc.
+
+With Storefront being a separate service (or set of services), should configuration settings in Magento Admin Panel be propagated to storefront service?
+If yes, in what way?
+
+## Configuration Propagation Guidelines
+
+This document covers general agreements for decisions about configuration propagation from Admin Panel to storefront. 
+A separate design decision should be made and documented for specific use cases.
+
+There are three use cases for the configuration, based on its impact.
+
+### 1. Configuration that impacts Admin Panel behavior
+
+Examples: admin ACLs that allow or deny parts of functionality for the admin user, reindex on update or by schedule.
+
+This type of configuration has nothing to do with Storefront and should not be propagated to storefront, as well as should not be exposed via export API. 
+
+### 2. Configuration that impacts data provided by Storefront
+
+Example: Base Media URL is used for calculation of media image URLs returned by Storefront API.
+
+This type of configuration should be propagated to Storefront in one of the following ways:
+
+1. Pre-calculate final data on back office side and provide final result to the Storefront during synchronization.
+   1. Pros: simpler implementation of Storefront due to eliminated necessity for Storefront to keep knowledge about additional configuration, including data calculation algorithms.
+   2. Cons: massive (up to full) reindexation necessary in case the configuration is changed.
+   
+To choose the right approach in a specific case, consider the following:
+
+1. How frequently the configuration is expected to change?
+   1. Frequent configuration changes leading every time to massive reindexation may be unacceptable.
+   2. Configuration changes expected a few times in the store lifespan may not be worth additional complexity on the Storefront side and full reindexation may be better in this case.
+2. Is it acceptable to have significant delay in data propagation after the configuration change?
+   1. Changing Base URL may be not a big issue, especially if a redirect can be setup. So it may be acceptable to have URLs to be fully updated in a few hours.
+   2. Changes in prices, on the other side, may not stand long delays.
+
+### 3. Configuration that impacts UI representation of Storefront data
+
+Example: include or exclude taxes in displayed prices. 
```

Comment:
My understanding was that we don't want to do price calculation in the catalog storefront service. Maybe message broker should be responsible for price calculation for price storefront service?

Reply:
It's not calculation, it's setting for presentation layer: do we show price including tax, excluding tax or both? I assume either both prices should be calculated by the back office (or whoever is responsible for it) and then client app just decides which price to use based on this config, or price calculation is synced with UI settings. I think 2nd option is too risky to follow, as for example a store may have multiple client apps (PC web site, mobile app, in-store app) and they may need to display prices differently. So I'd assume that price API should provide both incl and excl prices and prices UI config should say which of those should be displayed.